### PR TITLE
Fix PATH issues with swiftc and use swift to run directly in dev

### PR DESCRIPTION
**Pull request by airportyh**
## Why?

When installing swift to a repl. Compile + run fails if the .replit doesn't have Nix channel 22.11 specified because a different version of binutils is getting picked up by swiftc.

## Changes

1. Use swift directly to run a .swift program
2. For compiling swiftc, we create a wrapper program that freezes the PATH variable for that invocation of swiftc

## Testing

1. `nix-build -A replitPackages.modules`
2. Play with in at https://replit.com/@util/NixModulesExamples#swift.nix

### CODE REVIEW

File: modules/swift.nix [hunk 1]
```diff
@@ -1,26 +1,42 @@
...
-{
+let swiftc-wrapper = pkgs.stdenv.mkDerivation {
+  name = "swiftc-wrapper";
+  buildInputs = [pkgs.makeWrapper pkgs.swift];
+  src = ./.;
+
+  installPhase = ''
+    mkdir -p $out/bin
+    makeWrapper ${pkgs.swift}/bin/swiftc $out/bin/swiftc \
+      --set PATH ${pkgs.lib.makeBinPath [
```
Comment (ryantm) [inline modules/swift.nix:10]: Do you have to put swift on the path here, or is it sufficient to have an empty PATH?

Comment (ryantm) [reply modules/swift.nix:10]: It's weird to me that we need to have swift on the path of the swift compiler.

Comment (airportyh) [reply modules/swift.nix:10]: I think you are right. I'll try removing it.

Comment (airportyh) [reply modules/swift.nix:10]: Probably still want to set PATH though... to empty?

Comment (airportyh) [reply modules/swift.nix]: Yup. Setting the PATH to empty is needed to make it work.

**Pull request merged**